utils/buildAuthExchange.ts

```diff
@@ -0,0 +1,47 @@
+import { makeOperation } from "urql";
+import { authExchange } from "@urql/exchange-auth";
+
+function buildGetAuth(getToken: () => Promise<string>) {
+  const getAuth = async ({ authState }) => {
+    // TODO: We could start parsing and evaluating this token to avoid having to fetch it each time
+    const token = await getToken();
+
+    if (!authState) {
```

Comment:
Just asking for more context.  I'm confused about what this does if there is authState,  just return null?  Would that mean no token is provided?

Reply:
I am trying to wrap my head around it, also. This came directly from one of the examples. I need to dig into it more to better understand it myself.